jobs/base_model_validator.py

```diff
@@ -0,0 +1,192 @@
+# coding: utf-8
+#
+# Copyright 2021 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Beam PTransform and DoFn for base model validator functions"""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+import datetime
+import re
+
+import apache_beam as beam
+
+from core.domain import cron_services
+from core.platform import models
+from jobs import base_model_validator_errors as errors
+
+
+(base_models, user_models) = models.Registry.import_models(
+    [models.NAMES.base_model, models.NAMES.user])
+datastore_services = models.Registry.import_datastore_services()
+
+MAX_CLOCK_SKEW_SECS = datetime.timedelta(seconds=1)
+
+
+class BaseValidatorDoFn(beam.DoFn):
```

Comment:
I think it's OK to drop `DoFn` in the name, the `Validator` already implies that "validation" is the "function" being "done".